src/Helpers/CacheMap.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Backpack\Basset\Helpers;
+
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Storage;
+
+class CacheMap
+{
+    private $map = [];
+    private $path;
+    private $active = false;
+    private $dirty = false;
+
+    public function __construct()
+    {
+        $this->active = config('backpack.basset.cache_map', false);
+        $this->path = Storage::disk(config('backpack.basset.disk'))->path('.basset');
+
+        if (! $this->active) {
+            return;
+        }
+
+        // Load map
+        $this->map = File::exists($this->path) ? json_decode(File::get($this->path), true) : [];
+    }
+
+    /**
+     * Saves the cache map to the .basset file.
+     *
+     * @return void
+     */
+    public function save(): void
+    {
+        if (! $this->dirty || ! $this->active) {
+            return;
+        }
+
+        // save file
+        File::put($this->path, json_encode($this->map, JSON_PRETTY_PRINT));
+    }
+
+    /**
+     * Adds an asset to the cache map.
+     *
+     * @param  string  $asset
+     * @param  string  $path
+     * @return void
+     */
+    public function add(string $asset, string|bool $path = true): void
```

Comment:
Hmm... I don't think `$cacheMap->add()` is very expressive... 

In previous PRs I was suggesting we remove the `asset` from the function name, because the `asset` was already present in the class name, but in this case, when you write `$cacheMap->add()`... I don't think it's very clear what you're adding...

This is nitpicking of course but I think it'd read more clearly if it were:

```suggestion
    public function addAsset(string $asset, string|bool $path = true): void
```